### DISCUSSION TITLE
Tuya integrazione KO

### DISCUSSION THREAD

**Discussion by prinaldi54** [indomus/forum#933] [HUB: Home Assistant]
Buongiorno a tutti ho un problema con l'integrazione Tuya; dopo averla correttamente installata ha trovato tutti i mie devices. Il problema è che quando tento di azionare gli switch da Home Assistant si accende e dopo pochi secondi l'interfaccia lo segnala spento pur rimanendo acceso. Per spengerlo devo cliccare velocemente sullo switch ed allora lo spenge. Dall'App Tuya smart ovviamente tutto funziona perfettamente. La presa provvede anche a fornire i dati di assorbimento e voltaggio che nell'App si vedono perfettamente ed invece da Home Assistant,  pur esistendo i sensori, danno il valore zero ad eccezione di quello della tensione che da il valore rilevato.
Configurazione:
Home assistant core-2021.10.6
Tuya V2
Raspberry PI4 8GB

Qualcuno mi può aiutare?

Grazie Pierluigi

**Comment by poliant** [comment]
ciao Pierluigi,
prova a guardare nei log e controlla se c'è qualche errore docuto all'integrazione Tuya.

Io comunque ho avuto molti problemi con il componente di Home Assistant ed utilizzo le tuya tramite HACS ([qui](https://github.com/rospogrigio/localtuya)), senza passare dal cloud di tuya (ho HA core su container docker su RPi4)

ciao

**Reply by prinaldi54** [reply]
Naturalmente nei log è stata la prima cosa che ho controllato. Non compare niente. L'integrazione sembra partire perfettamente, riconosce tutti gli elementi che compaiono in Tuya smart ed indica addirittura il costruttore giusto.
Per quanto riguarda l'integrazione avevo capito (probabilmente male) che questa integrazione ufficiale era supportata da Tuya con le nuove API che andavano a sostituire le vecchie API non più mantenute. Inoltre, ed anche qui evidentemente mi sbagliavo, l'integrazione funzionava sia in locale che in cloud. Cosa ne pensi? Io in genere prediligo le versioni ufficiali che nel tempo dovrebbero dare più affidabilità, ma in questo caso non vedo molte soluzioni che ritornare all'integrazione non ufficiale.
Cioa

Pierluigi
